datadog_checks_base/datadog_checks/base/checks/db.py

```diff
@@ -20,3 +20,23 @@ def database_monitoring_metadata(self, raw_event: str):
 
     def database_monitoring_health(self, raw_event: str):
         self.event_platform_event(raw_event, "dbm-health")
+
+    @property
+    def reported_hostname(self) -> str | None:
+        raise NotImplementedError("reported_hostname is not implemented for this check")
```

Comment:
```suggestion
    @property
    @abstractmethod
    def reported_hostname(self) -> str | None:
        pass
```
An alternative to raising NotImplementedError would be to decorate these as abstractmethods (`from abc import abstractmethod`).  The benefit being unimplemented abstract methods will throw errors immediately upon class instantiation if missing vs the current approach which will lazily throw the error at first property access. Marking it abstractmethod also should get picked up in IDE's and linters I believe.